lib/jxl/encode.cc

```diff
@@ -2050,36 +2375,46 @@ void JxlEncoderCloseInput(JxlEncoder* enc) {
   JxlEncoderCloseFrames(enc);
   JxlEncoderCloseBoxes(enc);
 }
+
+JXL_EXPORT JxlEncoderStatus JxlEncoderFlushInput(JxlEncoder* enc) {
+  while (!enc->input_queue.empty()) {
```

Comment:
I guess we should. How to do this best? Can we initialize `JxlEncoderOutputProcessor`, so that it contains null pointers before `JxlEncoderSetOutputProcessor` was called or should we add something to the `JxlEncoder` struct? Like a bool 'using_output_processor`?

Reply:
We can just check if `external_output_processor_` is null or not.

Reply:
Ok, added a `OutputProcessorSet` with that check